Remove Declarations.yaml dependency from gen_autograd

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67497
* __->__ #67496

gen_autograd.py doesn't use `Declarations.yaml` any more, and removing
the dependency allows it to run in parallel with
`tools/codegen/gen.py`.

Differential Revision: [D32027251](https://our.internmc.facebook.com/intern/diff/D32027251)

peterbell10 committed:
Remove Declarations.yaml dependency from gen_autograd

gen_autograd.py doesn't use `Declarations.yaml` any more, and removing
the dependency allows it to run in parallel with
`tools/codegen/gen.py`.

[ghstack-poisoned]

peterbell10 committed:
Fix merge conflicts with master on "Remove Declarations.yaml dependency from gen_autograd"


gen_autograd.py doesn't use `Declarations.yaml` any more, and removing
the dependency allows it to run in parallel with
`tools/codegen/gen.py`.

[ghstack-poisoned]

albanD reviewed (commented):
Ho wow great catch!  when did that happen?

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32027251).

peterbell10 commented:
Yeah, I was expecting this to be a bit more work... Looks like #49735 did all the hard work but didn't update the build system.

dagitses commented on BUILD.bazel:
```diff
@@ -246,7 +246,7 @@ genrule(
         "aten/src/ATen/native/native_functions.yaml",
```
please remove the dependency from the srcs here

albanD replied:
Can this be done in a follow up? This just add un-necessary dependency to the bazel build right?

peterbell10 replied:
Done

dagitses replied:
correct this could have been done in a follow up

dagitses reviewed (changes requested):
looks like you missed the dep on the bazel rule

albanD commented:
Landing failed internally anyways...
@peterbell10 could you do the update suggested above please?

peterbell10 committed:
Update dependencies in BUILD.bazel on "Remove Declarations.yaml dependency from gen_autograd"


gen_autograd.py doesn't use `Declarations.yaml` any more, and removing
the dependency allows it to run in parallel with
`tools/codegen/gen.py`.

Differential Revision: [D32027251](https://our.internmc.facebook.com/intern/diff/D32027251)

[ghstack-poisoned]

albanD commented:
Thanks for the quick update!

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32027251).
